pkg/gofr/http/multipart_file_bind.go

```diff
@@ -146,13 +162,195 @@ func (uf *formData) setFieldValue(value reflect.Value, data string) (bool, error
 		return uf.setFloatValue(value, data)
 	case reflect.Bool:
 		return uf.setBoolValue(value, data)
-	case reflect.Invalid, reflect.Complex64, reflect.Complex128, reflect.Array, reflect.Chan, reflect.Func, reflect.Interface,
-		reflect.Map, reflect.Pointer, reflect.Slice, reflect.Struct, reflect.UnsafePointer:
-		// These types are not supported for setting via form data
+	case reflect.Slice, reflect.Array:
+		return uf.setSliceOrArrayValue(value, data)
+	case reflect.Interface:
+		return uf.setInterfaceValue(value, data)
+	case reflect.Struct:
+		return uf.setStructValue(value, data)
+	case reflect.Invalid, reflect.Complex64, reflect.Complex128, reflect.Chan, reflect.Func,
+		reflect.Map, reflect.Pointer, reflect.UnsafePointer:
 		return false, nil
+	}
+
+	return false, nil
+}
+
+func (uf *formData) setInterfaceValue(value reflect.Value, data string) (bool, error) {
+	// If the interface is not set to a concrete value, we can't modify it directly
+	if value.Kind() == reflect.Interface && value.IsNil() {
+		return false, errNilInterface
+	}
+
+	// If the value is a pointer to an interface, dereference it
+	if value.Kind() == reflect.Ptr && !value.IsNil() && value.Elem().Kind() == reflect.Interface {
+		value = value.Elem()
+	}
+
+	// If the interface holds a value, attempt to set the underlying type's value
+	if value.Kind() == reflect.Interface {
+		// Get the concrete value held by the interface
+		concreteValue := value.Elem()
+
+		// Try to set the concrete value using the underlying type
+		return uf.setFieldValue(concreteValue, data)
+	}
+
+	// If it's not an interface or the concrete value couldn't be set, return false
+	return false, fmt.Errorf("%w: %s", errUnsupportedInterfaceType, value.Kind())
+}
+
+func (uf *formData) setSliceOrArrayValue(value reflect.Value, data string) (bool, error) {
+	elemType := value.Type().Elem()
+
+	elements := strings.Split(data, ",")
+
+	// Create a new slice/array with appropriate length and capacity
+	var newSlice reflect.Value
+	if value.Kind() == reflect.Slice {
+		newSlice = reflect.MakeSlice(value.Type(), len(elements), len(elements))
+	} else if value.Kind() == reflect.Array {
+		if len(elements) > value.Len() {
+			return false, errDataLengthExceeds
+		}
+
+		newSlice = reflect.New(value.Type()).Elem() // Create a new zero-valued array
+	} else {
+		return false, fmt.Errorf("%w: %s", errUnsupportedKind, value.Kind())
+	}
```

Comment:
Using a switch on value.Kind would be easier to read

Reply:
replaced with a switch